### DISCUSSION TITLE
Evaluation

### DISCUSSION THREAD

**Discussion by mkutu** [precision-sustainable-ag/SemiF-AnnotationPipeline#33] [General]
@skovsens @savadikarc 

I'd like to get an understanding of accuracy for research purposes and maybe make this into a paper. I would like to do this and go your opinion on it. 

This would involve understanding the detection and segmentation accuracy of the pipeline.

What I'm thinking:
Create evaluation method for understanding
1. AutoSfM accuracy which we pretty much already have with the error csv output.
2. Bbox accuracy of model to detect individual plants or several plants/seedlings (eg several seedlings in one bounding box).
3. IoU of segmentation process

This requires developing a test set of images. 

* Detection accuracy can be calculated easily by looking at detection results and counting detected/missed plants. Individual images don't necessarily need to be annotated with bboxes. 
* Pixle-wise annotation, however, will need to be done to evaluate segmentation process. I'm thinking 10-50 images?

Ideally this would be done with a random selection of images of different species and at different growth stages.

**Comment by savadikarc** [comment]
@mkutu What is the theme that you want to address in the paper? 

When you say images do you mean 1 image containing multiple pots, or 1 image per plant? For the evaluation of bounding boxes and segmentation IoU, I think a significant number of images per species / per growth stage would make the results convicing, but I'm not sure how many images would be significant. @skovsens might be a better judge for that.

Another metric we could track is the accuracy of deduplication:
Acc_correct = number of pots deduplicated correctly / total number of pots
Acc_general = number of pots deduplicated / tota number of pots

Where correct deduplication is when we select the image directly overhead the pot

**Reply by mkutu** [reply]
@savadikarc good question about theme. Seeing as how I've spent and would like to continue spending more time on this project, it's important that I can get something from the effort (as I'm sure the same goes for you), so I'm looking for primary authorship (or lead co-author). This would obviously have a ton of people names attached to it.

I have a few ideas depending on your involvement and/or preference to include AutoSfM. I know the AutoSfM and other pieces of this are yours so I want to be flexible on what needs to be separated for authorship and attribution purposes especially on your end.  
*not only AutoSfM but other parts as well

So maybe a larger discussion on papers would be beneficial. I potentially see a couple options again depending on how involved you want to be.
Larger theme i had in mind involves developing a set of methods for building a large image repository that addresses the dataset gap in agriculture. Some structural options could include:
1. couple of papers about different parts of the annotation process
    -high-throughput vegetation segmentation and evaluation of multiple species under nursery conditions (likely publishing in agriculture- or plant science-based tech journal)
    - synthetic data generation for large image repository using semi-controlled plant nursery conditions under multiple growth stages (evaluate using segmentation model and test set of manually annotated images) 
        - This would be especially interesting because we could experiment with different combinations of synthetic data, train, and evaluate model performance.
        - i feel like this might have the best chance at comp sci or engineering journal???
2. A larger paper outlining pipeline, including segmentation, synthetic, training (evaluation), and iteration
    - With the larger goal of publishing in comp/engineering journal.

**Reply by mkutu** [reply]
also, 
for what it's worth, I think the AutoSfM could be it's own paper or built together into one of the others (the later of which would include shared co-authorship depending on how it's structured)

**Reply by savadikarc** [reply]
Makes sense for you to be the primary author I agree. Soren and you are the best judges for where AutoSfM fits in my opinion. We can discuss more in our meetings.

**Comment by mkutu** [comment]
> When you say images do you mean 1 image containing multiple pots, or 1 image per plant? 

Sorry is this referencing the evaluation?

> Another metric we could track is the accuracy of deduplication:
Acc_correct = number of pots deduplicated correctly / total number of pots
Acc_general = number of pots deduplicated / tota number of pots

Yes, which would go more into the impacts of AutoSfM

**Reply by savadikarc** [reply]
> Sorry is this referencing the evaluation?

Yes this is in reference to the evaluation. But thinking about it now, it should mean the image that is a part of the batch (row1_1, etc.). Nvm the question, I misunderstood the evaluation metrics.
